### PR TITLE
Delete vsphere-deployer, tf-deployer. Mark gcp-deployer deprecated.

### LINKED ISSUE #230: Delete gcp-deployer and tf-deployer directories
This issue tracks clusterctl being the primary and only bootstrapping mechanism, with all the functionality that is currently in gcp-deployer and tf-deployer.

### PR DESCRIPTION
**Pull request by karan**
**What this PR does / why we need it**: Deletes the now unnecessary deployer tools in favor of `clusterctl`. I've also updated documentation where needed.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

 #230

Not merging until after the community meeting on June 6.

/hold

### CODE REVIEW

**Comment by spew** [comment]
Thanks for opening this PR @karan. My thoughts are we should have feature parity in clusterctl before doing this (i.e. delete doesn't work yet)?

**Comment by karan** [comment]
In #230 we decided to make cluster deletion a little painful in the short-term, but using documentation instead. GCE in #261, vsphere in #286.

**Comment by karan** [comment]
Okay so we have decided to hold off on deleting gcp-deployer for a few more days until the existing integration test is migrated (@spew is working on that). This PR has been updated to delete only vsphere and tf deployers in the meantime.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ You can find an example implementation for GCE [here](https://github.com/kuberne
 
 ##### GCE Implementation
 
-For GCE, a [yaml file](https://github.com/kubernetes-sigs/cluster-api/blob/master/gcp-deployer/machine_setup_configs.yaml) holds the list of valid machine setup configs, 
+For GCE, a [config map](https://github.com/kubernetes-sigs/cluster-api/blob/6aecf9c80a1ca29b45cb43ebfd50ac0d57eb7132/clusterctl/examples/google/provider-components.yaml.template#L118) holds the list of valid machine setup configs,
```
[the remark this answers is not in the document]
Comment (karan) [reply CONTRIBUTING.md]: Yes. The template is changing a lot and line numbers move around quite a bit.

**Comment by karan** [comment]
Afaik, no one uses tf-deployer. The agreement was to not delete the provider for now, and others are free to work on it (including integrating with clusterctl).

**Comment by roberthbailey** [comment]
@dims expressed interest in extending the tf deployer to work with openstack but afaik hasn't started yet. I think we are saying that at this point he should build support into clusterctl instead.

**Comment by dims** [comment]
That's fair @roberthbailey thanks!

**Pull request merged**